tests/save_and_load_test.py

```diff
@@ -31,15 +33,15 @@ def test_picklability(inference_method, sampling_method: str, tmp_path):
 
     inference = inference_method(prior=prior)
     _ = inference.append_simulations(theta, x).train(max_num_epochs=1)
-    posterior = inference.build_posterior(sample_with=sampling_method).set_default_x(
-        x_o
-    )
+    posterior = inference.build_posterior(
+        sample_with=sampling_method, mcmc_parameters=mcmc_params_fast
+    ).set_default_x(x_o)
```

Comment:
I am not sure if this is supposed to be in this PR? I guess ideally we fix this in a separate PR but I think it's fine to keep it here.